packages/drivers/tinylicious-driver/src/insecureTinyliciousUrlResolver.ts

```diff
@@ -24,33 +22,49 @@ export const defaultTinyliciousEndpoint = "http://localhost";
  */
 export class InsecureTinyliciousUrlResolver implements IUrlResolver {
     private readonly fluidProtocolEndpoint: string;
+    private readonly tinyliciousEndpoint: string;
     public constructor(
-        private readonly tinyliciousPort = defaultTinyliciousPort,
-        private readonly tinyliciousEndpoint = defaultTinyliciousEndpoint,
-        ) {
-            this.fluidProtocolEndpoint = this.tinyliciousEndpoint.replace(/(^\w+:|^)\/\//, "fluid://");
-         }
+        port = defaultTinyliciousPort,
+        endpoint = defaultTinyliciousEndpoint,
+    ) {
+        this.tinyliciousEndpoint = `${endpoint}:${port}`;
+        this.fluidProtocolEndpoint = this.tinyliciousEndpoint.replace(/(^\w+:|^)\/\//, "fluid://");
+    }
 
     public async resolve(request: IRequest): Promise<IResolvedUrl> {
-        const url = request.url.replace(`${this.tinyliciousEndpoint}:${this.tinyliciousPort}/`, "");
+        if (request.headers && request.headers[DriverHeader.createNew] === true) {
+            return {
+                endpoints: {
+                    deltaStorageUrl: `${this.tinyliciousEndpoint}/deltas/tinylicious/new`,
+                    ordererUrl: this.tinyliciousEndpoint,
+                    storageUrl: `${this.tinyliciousEndpoint}/repos/tinylicious`,
+                },
+                id: "",
+                tokens: {},
+                type: "fluid",
+                url: `${this.fluidProtocolEndpoint}/tinylicious/new`,
+            };
+        }
```

Comment:
What is this code supporting enabling that wasn't there before?

Reply:
It handles a dedicated "create" request when the `createNew` header is provided. In this case, we don't need to parse the request url to determine the document ID

Reply:
I'll add a detailed comment.